parquet-column/src/main/java/org/apache/parquet/internal/column/columnindex/OffsetIndexBuilder.java

```diff
@@ -70,6 +72,14 @@ public long getFirstRowIndex(int pageIndex) {
     public int getPageOrdinal(int pageIndex) {
       return pageIndex;
     }
+
+    @Override
+    public Optional<Long> getUnencodedByteArrayDataBytes(int pageIndex) {
+      if (unencodedByteArrayDataBytes == null || unencodedByteArrayDataBytes.length == 0) {
```

Comment:
i'm not sure the comparison to 0 here is correct?  Can't a page be composed of entirely empty strings?

Reply:
Here the length means number of pages. If there is no page, then there is no unencodedByteArrayDataBytes either.